flash/video/classification/data.py

```diff
@@ -191,76 +189,54 @@ def _make_encoded_video_dataset(self, data) -> "LabeledVideoDataset":
-        ds = self._make_encoded_video_dataset(data)
+            data = list(zip(paths, [self.label_to_id[classname] for classname in labels]))
 
+        dataset = LabeledVideoDataset(
+            LabeledVideoPaths(data),
+            _make_clip_sampler(clip_sampler, clip_duration, clip_sampler_kwargs),
+            video_sampler=video_sampler,
+            decode_audio=decode_audio,
+            decoder=decoder,
+        )
         if self.training:
             self.set_state(LabelsState(self.id_to_label))
-            dataset.num_classes = len(self.labels_set)
-        return ds
+            self.num_classes = len(self.labels_set)
+        return dataset
 
 
-class VideoClassificationFiftyOneInput(
-    BaseVideoClassification,
-    FiftyOneInput,
-):
-    def __init__(
+class VideoClassificationFiftyOneInput(VideoClassificationInput):
+    def load_data(
         self,
-        clip_sampler: "ClipSampler",
+        sample_collection: SampleCollection,
+        clip_sampler: Union[str, "ClipSampler"] = "random",
+        clip_duration: float = 2,
+        clip_sampler_kwargs: Dict[str, Any] = None,
         video_sampler: Type[Sampler] = torch.utils.data.RandomSampler,
-        decode_audio: bool = True,
+        decode_audio: bool = False,
         decoder: str = "pyav",
         label_field: str = "ground_truth",
-    ):
-        super().__init__(
-            clip_sampler=clip_sampler,
-            video_sampler=video_sampler,
-            decode_audio=decode_audio,
-            decoder=decoder,
-        )
-        FiftyOneInput.__init__(
-            self,
-            label_field=label_field,
-        )
-
-    @property
-    def label_cls(self):
-        return fol.Classification
-
-    def _make_encoded_video_dataset(self, data: SampleCollection) -> "LabeledVideoDataset":
-        classes = self._get_classes(data)
+    ) -> "LabeledVideoDataset":
+        label_utilities = FiftyOneLabelUtilities(label_field, fol.Classification)
+        label_utilities.validate(sample_collection)
+        classes = label_utilities.get_classes(sample_collection)
         label_to_class_mapping = dict(enumerate(classes))
         class_to_label_mapping = {c: lab for lab, c in label_to_class_mapping.items()}
 
-        filepaths = data.values("filepath")
-        labels = data.values(self.label_field + ".label")
+        filepaths = sample_collection.values("filepath")
+        labels = sample_collection.values(label_field + ".label")
         targets = [class_to_label_mapping[lab] for lab in labels]
-        labeled_video_paths = LabeledVideoPaths(list(zip(filepaths, targets)))
-
-        ds: LabeledVideoDataset = LabeledVideoDataset(
-            labeled_video_paths,
-            self.clip_sampler,
-            video_sampler=self.video_sampler,
-            decode_audio=self.decode_audio,
-            decoder=self.decoder,
+
+        dataset = LabeledVideoDataset(
```

Comment:
Can we share this code? There is a lot of duplicated code.

Reply:
Not easily at the moment, but I think a further refactor / cleanup of the video data sources will be in order once the data pipeline V2 work is finished.